src/main/java/io/confluent/idesidecar/restapi/credentials/Credentials.java

```diff
@@ -44,6 +45,22 @@ enum Type {
     MUTUAL_TLS,
     OAUTH2,
     API_KEY_AND_SECRET,
+    SCRAM
+  }
+
+  enum HashAlgorithm {
+    SCRAM_SHA_256("SCRAM-SHA-256"),
+    SCRAM_SHA_512("SCRAM-SHA-512");
+
+    private final String value;
+
+    HashAlgorithm(String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
+    }
   }
```

Comment:
WDYT of moving the `HashAlgorithm` to the `ScramCredentials` record unless we need it for other credentials, too?

Also, what about removing the `SCRAM_` prefix from the values so that the values match the hash algorithm not the SCRAM mechanism? You could then introduce a method, like `toScramMechanism()` or similar, that returns the hash algorithm with the prefix.